### PR TITLE
Logs to cloudtrail

### PR DESCRIPTION
**Pull request by srinirei**
Control Plane Logging is added with log retention time of 180 days. 
Fargate logging includes both ALB and Nginx ingress controller log streams.
C Plugin (Cloudwatch_logs) used for shipping Fargate logs to Cloudwatch that does not support "log retention" parameter.

### CODE REVIEW

File: services/terraform/eks-provision.tf [hunk 1]
```diff
@@ -137,6 +129,36 @@ resource "aws_iam_role" "iam_role_fargate" {
   })
 }
 
+
+# -----------------------------------------------------------------------------------
+# Farage Logging Policy and Policy Attachment to the Fargage pod execution Iam role
```
Comment (mogul) [inline services/terraform/eks-provision.tf:134]:
```suggestion
# Fargate Logging Policy and Policy Attachment for the existing Fargate pod execution IAM role
```

File: services/terraform/eks-provision.tf [hunk 2]
```diff
@@ -189,6 +211,57 @@ resource "null_resource" "coredns_restart_on_fargate" {
   ]
 }
 
+# ---------------------------------------------------------------------------------------------
+# Fargate logging by fluentbit requires namespace aws-observability and Configmap
+# ---------------------------------------------------------------------------------------------
+
+# Configure Kubernetes namespace aws-observability with enabling aws-observability annotation. This
```
Comment (mogul) [inline services/terraform/eks-provision.tf:218]:
```suggestion
# Configure Kubernetes namespace aws-observability by adding the aws-observability annotation. This
```

File: services/terraform/eks-provision.tf [hunk 3]
```diff
@@ -189,6 +211,57 @@ resource "null_resource" "coredns_restart_on_fargate" {
...
+# annotation supported in terraform 0.13 or higher. So kubectl is used to provision the namespace.
```
Comment (mogul) [inline services/terraform/eks-provision.tf:219]:
```suggestion
# annotation is supported in terraform 0.13 or higher. So kubectl is used to provision the namespace.
```

File: services/terraform/eks-provision.tf [hunk 4]
```diff
@@ -189,6 +211,57 @@ resource "null_resource" "coredns_restart_on_fargate" {
...
+data "template_file" "logging" {
+  template = <<EOF
```
Comment (mogul) [inline services/terraform/eks-provision.tf:221]:
```suggestion
  template = <<-EOF
```
...then indent the heredoc below for readability.

File: services/terraform/eks-provision.tf [hunk 5]
```diff
@@ -137,6 +129,36 @@ resource "aws_iam_role" "iam_role_fargate" {
...
+      "Action": [
+        "logs:CreateLogStream",
+        "logs:CreateLogGroup",
+        "logs:DescribeLogStreams",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "*"
+    }]
+  }
+EOF
```
Comment (mogul) [inline services/terraform/eks-provision.tf:152]:
```suggestion
  EOF
```

**Review by mogul — CHANGES_REQUESTED** [review]
Minor wording fixes and indentation, then this looks good to go!

**Review by srinirei — COMMENTED** [review]
Thank you for the changes

File: services/terraform/eks-provision.tf [hunk 6]
```diff
@@ -189,6 +211,57 @@ resource "null_resource" "coredns_restart_on_fargate" {
...
 
+# ---------------------------------------------------------------------------------------------
+# Fargate logging by fluentbit requires namespace aws-observability and Configmap
+# ---------------------------------------------------------------------------------------------
+
+# Configure Kubernetes namespace aws-observability by adding the aws-observability annotation. This
+# annotation is supported in terraform 0.13 or higher. So kubectl is used to provision the namespace.
+data "template_file" "logging" {
+  template = <<-EOF
+kind: Namespace
```
Comment (mogul) [inline services/terraform/eks-provision.tf:222]: We changed from `<<EOF` to `<<-EOF` so you could indent this template... Can you do that?

**Review by mogul — DISMISSED** [review]

**Review by mogul — APPROVED** [review]

**Pull request merged**